axiom/tests/test_log_events.c

```diff
@@ -19,120 +20,138 @@
+  nr_log_event_set_priority(e, 0);
   return e;
 }
 
 static void test_events_success(void) {
   /* Normal operation */
-  nr_analytics_events_t* events = NULL;
+  nr_log_events_t* events = NULL;
   nr_log_event_t* e = NULL;
-  nr_random_t* rnd = nr_random_create_from_seed(12345);
-  const char* json;
+  void* test_e;
+  nr_vector_t* vector = NULL;
+  bool pass;
+  char* json;
   const char* expected;
 
   events = nr_log_events_create(10);
   tlib_fail_if_null("events created", events);
 
   e = create_sample_event(LOG_MESSAGE_0);
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 3);
+  nr_log_events_add_event(events, e);
 
   e = create_sample_event(LOG_MESSAGE_1);
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 2);
+  nr_log_events_add_event(events, e);
 
   e = create_sample_event(LOG_MESSAGE_2);
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 1);
+  nr_log_events_add_event(events, e);
 
   e = create_sample_event(LOG_MESSAGE_3);
-  nr_log_events_add_event(events, e, rnd);
-  nr_log_event_destroy(&e);
+  nr_log_event_set_priority(e, 0);
+  nr_log_events_add_event(events, e);
+
+  // minmax heap will store events like this (showing priorities)
+  //         0
+  //       3   2
+  //      1
+  //
+  // so when walking the heap and converting to a vector we expect
+  // the following order:
+  // LOG_MESSAGE_3 (pri 0)
+  // LOG_MESSAGE_0 (pri 3)
+  // LOG_MESSAGE_1 (pri 2)
+  // LOG_MESSAGE_2 (pri 1)
 
   tlib_pass_if_int_equal("events number seen updated", 4,
                          nr_log_events_number_seen(events));
   tlib_pass_if_int_equal("events number saved updated", 4,
                          nr_log_events_number_saved(events));
 
-#define TEST_EVENTS_GET(MESSAGE, IDX)                    \
-  do {                                                   \
-    json = nr_log_events_get_event_json(events, IDX);    \
-    tlib_fail_if_null("no json", json);                  \
-    expected = "[{" \
-                 "\"message\":\"" MESSAGE "\"," \
-                 "\"log.level\":\"" LOG_LEVEL "\"," \
-                 "\"timestamp\":" NR_STR2(LOG_TIMESTAMP) "," \
-                 "\"trace.id\":\"null\"," \
-                 "\"span.id\":\"null\"," \
-                 "\"entity.guid\":\"null\"," \
-                 "\"entity.name\":\"null\"," \
-                 "\"hostname\":\"null\"" \
-              "}]";                                    \
-    tlib_pass_if_str_equal("add event", expected, json); \
+  // convert log event pool to a vector and test
+  vector = nr_vector_create(10, NULL, NULL);
+  nr_log_events_to_vector(events, vector);
+
+#define TEST_EVENTS_GET(MESSAGE, IDX)                              \
+  do {                                                             \
+    pass = nr_vector_get_element(vector, IDX, &test_e);            \
```

Comment:
Same comment about `pass` name as in `cmd_txndata_transmit.c:nr_txndata_prepend_log_events` - `found` reads better.

Reply:
Fixed in commit [57d0c41d](https://github.com/newrelic/newrelic-php-agent/pull/515/commits/57d0c41d523951200221cc4a1dc83e877af15b0c).